### ISSUE TITLE
Accidentally closing Strapi window => losing unsaved work

### ISSUE THREAD

**Issue by antgel** [strapi/strapi#17731]
### Required System information

- Node.js version: 18.17.1
- NPM version: 9.8.1
- Strapi version: 4.12.4
- Database: sqlite
- Operating system: Ubuntu Linux 22.04
- Is your project Javascript or Typescript: N/A

### Describe the bug

I accidentally closed my Strapi browser window and lost an hour's work in progress. I was on my way to press Save but that's by the by.

### Steps to reproduce the behavior

- In Content Manager, work on a long edit to content in a Rich Text field.
- Close browser.
- Cry.
- Hit Ctrl-Shift-T, hoping that the form would contain the previous text.
- Cry again.
- Open issue here because surely this is a bug.

### Expected behavior

I expected a dialog or something similar, perhaps trapping `beforeunload` to stop me shooting myself in the foot.

### Additional context

Funnily enough I created https://feedback.strapi.io/feature-requests/p/autosave-when-editing-fields today because I could feel that something like this might happen.

Based on this, I can't use Strapi for editing content now; I'll have to edit elsewhere then paste in, which somewhat defeats the purpose of the Rich Text Field.

**Comment by antgel** [comment]
Adding to my OP: Hitting refresh in the browser suffers from the same problem.

**Comment by thakkarnetram** [comment]
Can I take up this issue & try ?

**Comment by Boegie19** [comment]
@thakkarnetram everyone can always make a PR. the way we see if someone is working on a issue is to check if a PR has been linked to this issue. Still writing a PR would not mean it gets accepted also this PR still this PR would be a back burner PR. so it could take a long time before it gets looked at and reviewed.

**Comment by thakkarnetram** [comment]
@Boegie19 Okay . so i will try and solve this issue and raise a PR linked with it !

**Comment by antgel** [comment]
@Boegie19 Just out of interest, why would this be a "back burner PR"? Losing unsaved work is quite a downer and disappointing UX.

**Comment by Boegie19** [comment]
@antgel since I agree it is an annoying issue. but there are issues with more impact that will get more priority. so prs like this can take a while to get accepted.

**Comment by antgel** [comment]
I don't know how "impact" is defined and I'm not underestimating the amount of work involved in reviewing PRs, but it's a shame that a ready offering might not get through to `main` in a reasonable time. :man_shrugging:

**Comment by Arunjenson** [comment]
any update on this?

**Comment by antgel** [comment]
No, apparently there are "issues with more impact" than being able to lose work at a keypress.

**Comment by oiorain** [comment]
Hi there! 

Thank you so much for reporting this. We're very sorry we didn't get to it faster. We're working really hard to catch up on issues on v5 and make the experience better. 
Please let us know if there are other small enhancements like this one that would make your lives easier.

This will be fixed on v5's next release. 

Have a lovely day

**Comment by antgel** [comment]
@oiorain @HichamELBSI Great that you finally got to this, thank you. 🙏

**Comment by antgel** [comment]
Not sure why the issue didn't close, but thanks again to the team for fixing this.
